### DISCUSSION TITLE
3.9.0 sound quality - frequency response

### DISCUSSION THREAD

**Discussion by iamthechrisb** [jamulussoftware/jamulus#2776] [Bug Investigation]
Jamulus team, thank you so much for the 3.9.0 release!  Is anyone noticing differences in the sound quality?
When I was first doing comparisons on the beta - I felt like lows sounded different, bun convinced myself they were not. Recently I've had people say that the highs sound different, or that the audio headroom sounds more compressed, dull maybe. I've looked through the release notes to see if there is something that looks like it could cause this.. but nothing standing out at me.

**Comment by pljones** [comment]
Several questions:
- What OS and OS version are you using?
- Which Jamulus package are you using (e.g. Windows ASIO vs Windows Jack or Mac OS vs Mac OS Native Silicon)?
- Does this happen only on specific Jamulus Servers?  If so, can you give examples?  Are there servers where it doesn't happen, again with examples?

Thanks!

**Reply by iamthechrisb** [reply]
Thank you. This appears to be server side. Various clients of different versions mac and windows.  Does not appear to happen on 3.8.2 server. Have seen this with Ubuntu 20.04 on arm and intel - compiled from source

**Reply by pljones** [reply]
Were you testing with 3.8.2 server and 3.9.0 server running on the same machine with the same configuration?  Can you supply the configuration details?

**Reply by iamthechrisb** [reply]
same machine for both versions, a few different machines. the configuration is pretty straightforward -F -T -P . On the 3.9.0 versions the api was running. Recording was started but not running on some of them.

**Reply by iamthechrisb** [reply]
update - This seems to really more apparent with a larger group playing on a server. With two different groups playing last night and A/B with 3.82. On 3.9.0 the room sounds more compressed?, there appears to be less headroom and the highs sound different..

**Reply by hoffie** [reply]
Hrm, I can't think of any current audio-quality related changes. One way to narrow this problem down would be bisect-ing. This would require someone who can reproduce the problem to build and test with different intermediate versions of Jamulus.

It's basically:
```
git bisect start
git bisect good r3_8_2
git bisect bad r3_9_0
make clean
make
make install # Or use ./Jamulus directly
```

If the problem exists in this build, run `git bisect bad`. If it doesn't, run `git bisect good`.
After that, run `make clean; make; make install` again and repeat the good/bad/compile dance. Once done (it'll likely need about 8 such steps), git will output the likely culprit...

@iamthechrisb Could you do that?

**Reply by iamthechrisb** [reply]
@hoffie Thanks -- I'll see if I can get this done this week. If anyone is close to US eastern time and wants to hear with a larger group, stop in and visit us - I may be able to flip back and fourth for a comparison

**Reply by iamthechrisb** [reply]
@hoffie Sorry I havent been able to get to this -- part of that is not being sure I know what I am doing.. 

Is it possible that the sound quality I am hearing is related to #2806 ?

**Reply by hoffie** [reply]
> Sorry I havent been able to get to this -- part of that is not being sure I know what I am doing..

If you've got any questions about sub-steps, feel free to ask. :)

> Is it possible that the sound quality I am hearing is related to #2806 ?

As far as I understood, you are not using a Mac M1, are you? If so, there would be a very small chance that this (unfixed) issue is related, but I don't think so.

**Reply by iamthechrisb** [reply]
not mac M1 -- but arm

**Reply by hoffie** [reply]
Ah, right. I don't think anything has changed in this regard though. As said, the above mentioned issue has not been worked on yet, so there have been no changes in this regard.
Still hoping for a `git bisect` to explain this :)
